apps/full-stack-tests/src/components/properties/PresentationFilterBuilderValueRenderer.test.tsx

```diff
@@ -0,0 +1,261 @@
+          </ECEntityClass>
+        `,
+      );
+      schemaAlias = schema.schemaAlias;
+      const physicalModel = insertPhysicalModelWithPartition({ builder, codeValue: "TestPhysicalModel" });
+      const category = insertSpatialCategory({ builder, codeValue: "Test SpatialCategory" });
+      const parentElement = insertPhysicalElement({
+        builder,
+        modelId: physicalModel.id,
+        categoryId: category.id,
+        classFullName: schema.items.MyPhysicalObjectParent.fullName,
+        userLabel: "Parent Test Class",
+        ["PropertyName"]: "Parent",
+      });
+      const element1 = insertPhysicalElement({
+        builder,
+        modelId: physicalModel.id,
+        categoryId: category.id,
+        classFullName: schema.items.MyPhysicalObject1.fullName,
+        parentId: parentElement.id,
+        userLabel: "Test Class",
+        ["PropertyName"]: "Value1",
+      });
+      const element2 = insertPhysicalElement({
+        builder,
+        modelId: physicalModel.id,
+        categoryId: category.id,
+        classFullName: schema.items.MyPhysicalObject2.fullName,
+        parentId: parentElement.id,
+        userLabel: "Test Class 2",
+        ["PropertyName"]: "Value2",
+      });
+      return { parentElement, element1, element2, category };
+    });
+
+    const testProperty = {
+      name: `#pc_${schemaAlias}_MyPhysicalObjectParent_PropertyName`,
+      displayLabel: "PropertyName",
+      typename: "string",
+    };
+
+    const keys = new KeySet([
+      { id: imodel.element1.id, className: imodel.element1.className },
+      { id: imodel.element2.id, className: imodel.element2.className },
+    ]);
+
+    const testDescriptor = await Presentation.presentation.getContentDescriptor({
+      imodel: imodel.imodel,
+      rulesetOrId: {
+        id: `Test descriptor ruleset`,
+        rules: [
+          {
+            ruleType: "Content",
+            specifications: [
+              {
+                specType: "SelectedNodeInstances",
+              },
+            ],
+          },
+        ],
+      },
+      displayType: DefaultContentDisplayTypes.PropertyPane,
+      keys,
+    });
+
+    if (testDescriptor === undefined) {
+      expect(false);
+      return;
+    }
+
+    const selectedClasses: ClassInfo[] = [
+      {
+        id: imodel.element1.id,
+        name: imodel.element1.className.replace(".", ":"),
+        label: "Test Class",
+      },
+    ];
+
+    const { baseElement, getByRole } = render(
+      <PresentationFilterBuilderValueRenderer
+        property={testProperty}
+        onChange={() => {}}
+        imodel={imodel.imodel}
+        descriptor={testDescriptor}
+        selectedClasses={selectedClasses}
+        operator={"is-equal"}
+      />,
+    );
+
+    // trigger loadTargets function
+    const user = userEvent.setup();
+    const combobox = await waitFor(() => getByRole("combobox"));
+
+    await waitFor(async () => {
+      await user.click(combobox);
```

Comment:
Is there a reason we click is done in `waitFor`?

Reply:
Leaving outside the waitFor seems to cause an act warning. Moved it to the inside to avoid it flooding the console.

Reply:
user-event library should add act wrappers to avoid act warnings.

Reply:
Changed